rtt/internal/BindStorage.hpp

```diff
@@ -64,27 +64,29 @@ namespace RTT
         template<class T>
         struct AStore
         {
+            typedef T arg_type;
             T arg;
             AStore() : arg() {}
             AStore(T t) : arg(t) {}
             AStore(AStore const& o) : arg(o.arg) {}
 
             T& get() { return arg; }
             void operator()(T a) { arg = a; }
-            operator T() { return arg;}
+            operator T&() { return arg; }
```

Comment:
@psoetens Was there a good reason why the operator did not return a reference in the original code? I think not: The object is already copied at least once when creating the `AStore<T>` instance or via `operator()(T a)`. So no reason to copy it again to read it. For references we use the specialization below instead.

Even without the reference return type the object was not copied for the actual operation call because before this patch `.get()` was used to access it in `BindStorageImpl<ToBind>` below.

Reply:
I remember that I tested this code with objects which did a cout in a copy constructor to count the number of copies made, and then introduced references to reduce the number of copies. That said, I don't remember if there was a formal place in the architecture where a copy 'must' be made, but the AStore seems to be the best place to do it, if required. AStore was meant to mimic the classical C++ function argument behavior.